src/components/widgets-settings/organizations/organizations.directive.coffee

```diff
@@ -21,9 +34,20 @@ module.controller('SettingOrganizationsCtrl', ($scope, $log, ImpacDashboardsSvc)
     ImpacDashboardsSvc.load().then(
       (config) ->
         $scope.dashboardOrganizations = config.currentDashboard.data_sources
+
         if w.metadata? && w.metadata.organization_ids?
-          for org in $scope.dashboardOrganizations
-            w.selectedOrganizations[org.uid] = _.contains(w.metadata.organization_ids, org.uid)
+          widgetOrgIds = w.metadata.organization_ids
+          if singleOrgMode()
+            resetSelectedOrgs()
+            if _.map(widgetOrgIds, -> true).length > 1
+              currentOrganization = ImpacMainSvc.config.currentOrganization
+              w.selectedOrganizations[currentOrganization.uid] = true
+            else
+              w.selectedOrganizations[widgetOrgIds[0]] = true
+
+          if multiOrgMode()
```

Comment:
Maybe use `else if` like you have in the `$scope.toggleSelectOrganization` method.